### PR TITLE
Minor "topics-map" adjustments

### PR DESCRIPTION
**Pull request by pypt**
Just a few very cosmetic adjustments for `topics-map`:

* Use curl instead of wget as it's preinstalled, well maintained and supports all of the fancy protocols
* Tabs -> spaces
* Do `apt-get -y --no-install-recommends install` instead of just `apt-get -y install` to avoid installing docs, samples and such
* Store `forceatlas2.jar` and `gephi-toolkit.jar` without a hardcoded version number
* Remove `nice` as that won't much (if anything) in a container
* Call to `connect_to_db()` only once as we'll reconnect on every `run_job()` anyway, and there's no point to disconnect from PgBouncer nicely - we can just cut the connection
* Replace module monkey patching (`module.submodule.CONSTANT = 'test value'`) with a simple parameter with a default value -- easier to understand and will work better with Perl (not that this code will be called from Perl, but it's a good pattern to stick by)
* Install Java first, then the JARs, and then do the rest. `Dockerfile` should start with "heavier" steps that don't change too often (e.g. install some sort of a runtime), and end with constantly changing steps (e.g. update worker's source code) so that Docker can skip rebuilding heavy parts whenever it can. Installing Java before JARs is something like 2% more effective :)
* Instead of hardcoding `-Xmx8g`, read memory limit allotted to the container using `/container_memory_limit.sh` and pass 90% of the limit as `-Xmx`. Container's memory limit might be smaller than 8 GB (host machine might not even have 8 GB available), and it's nice to be able to set this limit without having to edit the source code and rebuild container image.
* Add `topics-map` to `docker-compose.dist.yml` so that we'll know how to run it. IMHO it lacks a sensible description, reasonable worker count and practical RAM limit though.

### CODE REVIEW

**Comment by hroberts** [comment]
Thanks Linas! 

These look great. A couple of small responses below.

> * Remove `nice` as that won't much (if anything) in a container

I thought that the nice would push up into the host, but I see that's not the case.  It looks like it is possible to setup a container so that this would be the case, but it's probably not worth the effort unless we notice this becoming a problem.  The challenge is that map generation is quite cpu intensive for very large topics, so the election topic can take about 10 minutes to run, and during that time it is sitting on 100% of 8 cpus.  Nice'ing those jobs lets me tell the system to use all the CPU if it is idle but be nice otherwise, which is very different than only ever being able to use 4 CPUs or whatever.  For the time being, I'm fine with just letting the container hug the 8 CPUs at will, and we can fix it later if it becomes a problem


> * Replace module monkey patching (`module.submodule.CONSTANT = 'test value'`) with a simple parameter with a default value -- easier to understand and will work better with Perl (not that this code will be called from Perl, but it's a good pattern to stick by)

much better.  thanks!

> * Install Java first, then the JARs, and then do the rest. `Dockerfile` should start with "heavier" steps that don't change too often (e.g. install some sort of a runtime), and end with constantly changing steps (e.g. update worker's source code) so that Docker can skip rebuilding heavy parts whenever it can. Installing Java before JARs is something like 2% more effective :)

ahhh, didn't think of that.  good to know.

> * Instead of hardcoding `-Xmx8g`, read memory limit allotted to the container using `/container_memory_limit.sh` and pass 90% of the limit as `-Xmx`. Container's memory limit might be smaller than 8 GB (host machine might not even have 8 GB available), and it's nice to be able to set this limit without having to edit the source code and rebuild container image.
> * Add `topics-map` to `docker-compose.dist.yml` so that we'll know how to run it. IMHO it lacks a sensible description, reasonable worker count and practical RAM limit though.

I have a small worker count and high memory and cpu requirements so that we can process large maps.  The large maps will just fail without the high memory, and they will take forever without the high cpus.

**Comment by hroberts** [comment]
everything look good to merge other than the draw graph_format

**Comment by pypt** [comment]
> I have a small worker count and high memory and cpu requirements so that we can process large maps. The large maps will just fail without the high memory, and they will take forever without the high cpus.

That's alright, the improvement now is that you can adjust those in production's `docker-compose.yml` (https://github.com/berkmancenter/mediacloud/blob/master/apps/docker-compose.dist.yml#L1953-L1956) or Portainer.

File: apps/topics-map/src/python/topics_map/map.py [hunk 1]
```diff
@@ -454,91 +473,95 @@ def draw_graph(graph, graph_format='svg'):
         alpha=0.7
     )
 
-    draw_labels(graph)
-        
+    draw_labels(graph=graph)
+
     plt.axis('off')
 
-    if graph_format == 'draw':
```
[the remark this answers is not in the document]
Comment (pypt) [reply apps/topics-map/src/python/topics_map/map.py]: No problem, reverted in `master`.

**Comment by pypt** [comment]
> I thought that the nice would push up into the host, but I see that's not the case.

With a supposedly better load distribution across our servers, most of them at least 50% idle now at any given moment so I'm not sure if we even need to get into process prioritization just yet - if a process needs some CPUs, it can just get it right away.

If we find ourselves needing to let some processes have preference over others, that can be configured with `docker-compose.yml` itself with more exotic `resources/(cpu_shares|cpu_quota|cpuset)` options, but I don't think we're there yet - after all, we have 352 vCPUs and 2.22 TB of RAM across our servers so that's plenty of playground for everyone!

**Pull request merged**